evaluate/task/migrate_test.go

```diff
@@ -0,0 +1,332 @@
+				"file.go",
+				"file_test.go",
+			},
+			FilePath: "file_test.go",
+
+			ExpectedFilePaths: []string{
+				"file.go",
+				"file_test.go",
+			},
+		})
+		validate(t, &testCase{
+			Name: "Multiple",
+
+			Language: &golang.Language{},
+			AllFilePaths: []string{
+				"fileA.go",
+				"fileA_test.go",
+				"fileB.go",
+				"fileB_test.go",
+				"fileC.go",
+				"fileC_test.go",
+			},
+			FilePath: "fileB_test.go",
+
+			ExpectedFilePaths: []string{
+				"fileB.go",
+				"fileB_test.go",
+			},
+		})
+	})
+	t.Run("Java", func(t *testing.T) {
+		validate(t, &testCase{
+			Name: "Single",
+
+			Language: &java.Language{},
+			AllFilePaths: []string{
+				filepath.Join("src", "main", "java", "com", "eval", "File.java"),
+				filepath.Join("src", "test", "java", "com", "eval", "FileTest.java"),
+			},
+			FilePath: filepath.Join("src", "test", "java", "com", "eval", "FileTest.java"),
+
+			ExpectedFilePaths: []string{
+				filepath.Join("src", "main", "java", "com", "eval", "File.java"),
+				filepath.Join("src", "test", "java", "com", "eval", "FileTest.java"),
+			},
+		})
+		validate(t, &testCase{
+			Name: "Multiple",
+
+			Language: &java.Language{},
+			AllFilePaths: []string{
+				filepath.Join("src", "main", "java", "com", "eval", "FileA.java"),
+				filepath.Join("src", "main", "java", "com", "eval", "FileB.java"),
+				filepath.Join("src", "main", "java", "com", "eval", "FileC.java"),
+				filepath.Join("src", "test", "java", "com", "eval", "FileATest.java"),
+				filepath.Join("src", "test", "java", "com", "eval", "FileBTest.java"),
+				filepath.Join("src", "test", "java", "com", "eval", "FileCTest.java"),
+			},
+			FilePath: filepath.Join("src", "test", "java", "com", "eval", "FileBTest.java"),
+
+			ExpectedFilePaths: []string{
+				filepath.Join("src", "main", "java", "com", "eval", "FileB.java"),
+				filepath.Join("src", "test", "java", "com", "eval", "FileBTest.java"),
+			},
+		})
+	})
+}
+
+func TestValidateMigrateRepository(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		Before func(repositoryPath string)
+
+		ExpectedError func(t *testing.T, err error)
+	}
+
+	validateJava := func(t *testing.T, tc *testCase) {
+		validateRepository := &tasktesting.TestCaseValidateRepository{
+			Name: tc.Name,
+
+			Before: tc.Before,
+
+			TestdataPath:   filepath.Join("..", "..", "testdata"),
+			RepositoryPath: filepath.Join("java", "migrate-plain"),
+			Language:       &java.Language{},
+
+			ExpectedError: tc.ExpectedError,
+		}
+
+		validateRepository.Validate(t, validateMigrateRepository)
+	}
+
+	removeFilesWithCondition := func(language language.Language, repositoryPath string, condition func(filePath string) bool) {
+		_, logger := log.Buffer()
+
+		filePaths, err := language.Files(logger, repositoryPath)
+		require.NoError(t, err)
+
+		for _, filePath := range filePaths {
+			if condition(filePath) {
+				require.NoError(t, os.Remove(filepath.Join(repositoryPath, filePath)))
+			}
+		}
+	}
```

Comment:
nitpick: would maybe not use a helper here, just delete the files that you don't need one by one, cause this adds another complication layer :thinking: